drivers/spi/spi_sam0.c

```diff
@@ -312,27 +312,26 @@ static void spi_sam0_fast_transceive(struct spi_config *config,
  * - Zero or more trailing RX only bufs
  * - Zero or more trailing TX only bufs
  */
-static bool spi_sam0_is_regular(const struct spi_buf *tx_bufs,
-				size_t tx_count, struct spi_buf *rx_bufs,
-				size_t rx_count)
+static bool spi_sam0_is_regular(const struct spi_buf_array *tx_bufs,
+				const struct spi_buf_array *rx_bufs)
 {
-	while (tx_count != 0 && rx_count != 0) {
-		if (tx_bufs->len != rx_bufs->len) {
+	while (tx_bufs->count != 0 && rx_bufs->count != 0) {
+		if (tx_bufs->bufs->len != rx_bufs->bufs->len) {
 			return false;
 		}
 
-		tx_bufs++;
-		tx_count--;
-		rx_bufs++;
-		rx_count--;
+		tx_bufs->bufs++;
```

Comment:
Does it compile? Is should not, since `tx_bufs` points to a constant structure.

Reply:
well it does. I overlooked that. my compiler is missing something here.

anyway bufs it spi_buf_array cannot be const, as it may vary at runtime (depends on the logic).

Reply:
In my opinion it should be `const`. There is no need for a driver to modify the provided structures. You can use local variables for iterating through the pointed data and counting down. It could be more effective as it could be done in CPU registers, and now you're modifying memory.